packages/instrumentation-openai/src/instrumentation.ts

```diff
@@ -0,0 +1,748 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// avoids a dependency on @opentelemetry/core for hrTime utilities
+import { performance } from 'perf_hooks';
+
+import { context, SpanKind, SpanStatusCode, trace } from '@opentelemetry/api';
+import type { Attributes, Context, Histogram, Span } from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+} from '@opentelemetry/instrumentation';
+import { SeverityNumber } from '@opentelemetry/api-logs';
+
+import {
+  ATTR_EVENT_NAME,
+  ATTR_GEN_AI_OPERATION_NAME,
+  ATTR_GEN_AI_REQUEST_ENCODING_FORMATS,
+  ATTR_GEN_AI_REQUEST_FREQUENCY_PENALTY,
+  ATTR_GEN_AI_REQUEST_MAX_TOKENS,
+  ATTR_GEN_AI_REQUEST_MODEL,
+  ATTR_GEN_AI_REQUEST_PRESENCE_PENALTY,
+  ATTR_GEN_AI_REQUEST_TOP_P,
+  ATTR_GEN_AI_RESPONSE_FINISH_REASONS,
+  ATTR_GEN_AI_RESPONSE_ID,
+  ATTR_GEN_AI_RESPONSE_MODEL,
+  ATTR_GEN_AI_SYSTEM,
+  ATTR_GEN_AI_TOKEN_TYPE,
+  ATTR_GEN_AI_USAGE_INPUT_TOKENS,
+  ATTR_GEN_AI_USAGE_OUTPUT_TOKENS,
+  EVENT_GEN_AI_ASSISTANT_MESSAGE,
+  EVENT_GEN_AI_CHOICE,
+  EVENT_GEN_AI_SYSTEM_MESSAGE,
+  EVENT_GEN_AI_TOOL_MESSAGE,
+  EVENT_GEN_AI_USER_MESSAGE,
+  METRIC_GEN_AI_CLIENT_OPERATION_DURATION,
+  METRIC_GEN_AI_CLIENT_TOKEN_USAGE,
+} from './semconv';
+import { PACKAGE_NAME, PACKAGE_VERSION } from './version';
+import { getEnvBool, getAttrsFromBaseURL } from './utils';
+import { OpenAIInstrumentationConfig, GenAI } from './types';
+
+// Use `DEBUG=elastic-opentelemetry-instrumentation-openai ...` for debug output.
+// Or use `node --env-file ./dev.env ...` in this repo.
+import createDebug from 'debug';
+const debug = createDebug('elastic-opentelemetry-instrumentation-openai');
+(debug as any).inspectOpts = { depth: 50, colors: true };
+
+class OpenAIInstrumentation extends InstrumentationBase<OpenAIInstrumentationConfig> {
```

Comment:
you need to export the class otherwise the compiler does generates an empty type definition file.

```js
export {};
//# sourceMappingURL=instrumentation.d.ts.map
```

```suggestion
export class OpenAIInstrumentation extends InstrumentationBase<OpenAIInstrumentationConfig> {
```

Reply:
Thanks, fixed in commit efb119cd96b3040cdbb12fcf132389688a4eafb8 I believe.